tests/cache/test_decorator.py

```diff
@@ -586,3 +586,29 @@ def generate(*args):
 
         generate.set({7: 18, 10: 15})
         eq_(generate(2, 7, 10), ['2 5', 18, 15])
+
+    def test_cache_preserve_sig(self):
+        reg = self._region()
```

Comment:
why reg and not `region`?

Reply:
oh I guess that's convention. well then.

Reply:
:)